### DISCUSSION TITLE
How can I manually install and set up the checkstyle linter for Vim and ALE?

### DISCUSSION THREAD

**Discussion by TheButterflyMon** [w0rp/ale#4082] [Q&A - Ask for help with problems] [answered]
I'm trying to lint `*.java` files with Vim and ALE. It works fine when I use `javac` as the linter, but I would like to use checkstyle. I added the following to my `.vimrc` to tell ALE to only use the checkstyle linter for `*.java` files:
```
 " for Java files, use 'checkstyle' (and only checkstyle)
 let g:ale_linters = {
       \ 'java': ['checkstyle'],
       \ }
```
I have an old MacBook with OSX Yosemite, so I cannot use Homebrew to download and install checkstyle as some of the required components (like Go) will not install on my machine. I downloaded `checkstyle-9.3-all.jar`, renamed it to `checkstyle.jar`, and put it in a folder `/XXX/checkstyle/checkstyle.jar`. Then, I updated the `$CLASSPATH` and `$PATH` variables in my `.profile`:
```
export CLASSPATH=/XXX/checkstyle/checkstyle.jar:$CLASSPATH
export PATH=/XXX/checkstyle:$PATH
```
Finally, I updated the global variable `g:ale_java_checkstyle_executable` in my `.vimrc`:
```
" see :help g:ale_java_checkstyle_config
let g:ale_java_checkstyle_executable = 'checkstyle.jar'
```
I can use checkstyle from the command line with `java com.puppycrawl.tools.checkstyle.Main` ( `java com.puppycrawl.tools.checkstyle.Main --help` shows me the help screen ). I can also display the help information with `java -jar checkstyle.jar --help`. But for some reason, I have to invoke that command from the folder which contains `checkstyle.jar`; otherwise I get an Error: `Error: Unable to access jarfile checkstyle.jar`.

However, ALE's linting does not work with checkstyle. I searched all over the Internet and tried many different variations, but nothing worked. I've attached a screenshot of the `:ALEInfo` command. If anyone knows what I am doing wrong, any help is appreciated.

![vim_ale_checkstyle](https://user-images.githubusercontent.com/12036466/154483923-3e676a7a-3689-4dea-affd-fef14af8adec.jpg)

**Comment by hsanson** [comment, marked as answer]
ALE requires a binary called "checkstyle" to be present on your PATH in order to find it and use it. Simplest way to get it to work on your environment is to create an executable bash script named "checkstyle" that calls whatever command needed to run the jar:

```sh
#!/bin/sh

java -jar /path/to/checkstyle.jar com.puppycrawl.tools.checkstyle.Main "$@"
```

Copy the bash script into your path (e.g. /usr/local/bin/checkstyle) and make it executable (e.g. chmod +x /usr/local/bin/checkstyle). Note you will have to ensure the script accepts and passes any arguments directly (e..g "$@").

**Reply by TheButterflyMon** [reply]
Thank you. That works perfectly!
